splink/diagnostics.py

```diff
@@ -0,0 +1,158 @@
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.session import SparkSession
+from pyspark.context import SparkContext, SparkConf
+from pyspark.sql import SQLContext
+from pyspark.sql.types import StructType, StructField, StringType, DoubleType, Row
+import pyspark.sql.functions as f
+from pyspark.sql.functions import when
+from .check_types import check_types
+
+
+import pandas as pd
+import warnings
+
+
+altair_installed = True
+try:
+    import altair as alt
+except ImportError:
+    altair_installed = False
+
+
+hist_def_dict = {
+    "$schema": "https://vega.github.io/schema/vega-lite/v4.8.1.json",
+    "config": {
+        "title": {"fontSize": 14},
+        "view": {"continuousHeight": 300, "continuousWidth": 400},
+    },
+    "data": {"values": None},
+    "encoding": {
+        "tooltip": [{"field": "count_rows", "title": "count", "type": "quantitative"}],
+        "x": {
+            "axis": {"title": "splink score"},
+            "bin": "binned",
+            "field": "splink_score_bin_low",
+            "type": "quantitative",
+        },
+        "x2": {"field": "splink_score_bin_high"},
+        "y": {
+            "field": "normalised",
+            "type": "quantitative",
+            "axis": {"title": "Probability density"},
+        },
+        "height": 200,
+        "mark": "bar",
+        "title": "Estimated Probability Density",
+        "width": 700,
+    },
+}
+
+
+@check_types
+def _splink_score_histogram(
+    df_e: DataFrame, spark: SparkSession, splits=None, score_colname=None,
+):
+
+    """splink score histogram diagnostic 
+    
+    
+        Args:
+    
+        
+            df_e (DataFrame): A dataframe of record comparisons containing a splink score, 
+            e.g. as produced by the .... function
+            
+            spark (SparkSession): SparkSession object
+            
+            score_colname : is the score in another column? defaults to None
+            
+            splits : list of splits for histogram bins. Has [0.05, 0.1, 0.25, 0.5, 0.75, 0.9, 0.95] for default
```

Comment:
https://github.com/moj-analytical-services/splink/pull/139#discussion_r533244221

I still think we should mirror the pyspark [histogram API](https://spark.apache.org/docs/1.1.1/api/python/pyspark.rdd.RDD-class.html#histogram) here, using the same param name and description.

I think now we're using a PMF the default should be a lot more granular splits, say 100

I would use the same API as provided by the histogram function to be consistent. i.e a param of buckets which takes either an int or a list of splits. Probably default should be int (equal width buckets) and probably quite a lot (say, 100) and link off to the Spark docs for the detailed explanation